src/utils/cst_tokenstream.c

```diff
@@ -408,44 +601,119 @@ int ts_get_stream_size(cst_tokenstream *ts)
         return 0;
 }
 
-cst_string private_ts_getc(cst_tokenstream *ts)
+void private_ts_getc(cst_tokenstream *ts)
 {
```

Comment:
This may be outside of the scope of the PR, but I don't see `private_ts_getc()` used anywhere. Can it be removed all together? It seems to expose `internal_ts_get()`but that makes the name quite counter intuitive.